packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Models/Types/MrwSerializationTypeProvider.cs

```diff
@@ -152,11 +235,99 @@ internal CSharpMethod BuildIModelGetFormatFromOptionsMethod()
             // ModelReaderWriterFormat IPersistableModel<T>.GetFormatFromOptions(ModelReaderWriterOptions options)
             return new CSharpMethod
             (
-              new MethodSignature(nameof(IPersistableModel<object>.GetFormatFromOptions), null, null, MethodSignatureModifiers.None, typeof(string), null, new[] { SerializationOptionsParameter }, ExplicitInterface: _iPersistableModelTInterface),
+              new MethodSignature(nameof(IPersistableModel<object>.GetFormatFromOptions), null, null, MethodSignatureModifiers.None, typeof(string), null, new[] { _serializationOptionsParameter }, ExplicitInterface: _iPersistableModelTInterface),
               jsonWireFormat
             );
         }
 
+        /// <summary>
+        /// Builds the serialization constructor for the model.
+        /// </summary>
+        /// <returns>The constructed serialization constructor.</returns>
+        internal CSharpMethod BuildSerializationConstructor()
+        {
+            var serializationCtorParameters = BuildSerializationConstructorParameters();
+
+            return new CSharpMethod(
+                signature: new ConstructorSignature(
+                    Type,
+                    $"Initializes a new instance of {Type:C}",
+                    null,
+                    MethodSignatureModifiers.Internal,
+                    serializationCtorParameters),
+                bodyStatements: new MethodBodyStatement[]
+                {
+                    GetPropertyInitializers(serializationCtorParameters)
+                });
+        }
+
+        private MethodBodyStatement GetPropertyInitializers(IReadOnlyList<Parameter> parameters)
+        {
+            List<MethodBodyStatement> methodBodyStatements = new();
+
+            foreach (var param in parameters)
+            {
+                if (param.Name == _rawDataField?.Name.ToVariableName())
+                {
+                    methodBodyStatements.Add(Snippet.Assign(new MemberExpression(null, _rawDataField.Name), new ParameterReferenceSnippet(param)));
+                    continue;
+                }
+
+                ValueExpression initializationValue = new ParameterReferenceSnippet(param);
```

Comment:
The init value will be different for collections.  We need to see if the value passed in is null and if so init to one of the changetracking types.

This is where I think CSharpType should have a property called `public CSharpType InitializationType { get; }` which would return the type to initialize.

OR

A parameter needs to have a property which returns its various init expressions.  I do think there would be different expressions depending on whether it is a ctor or a method which might be one point in favor of the csharptype.

Either way this method needs to consume this and construct the appropriate expression for a regular param vs a collection param.

Reply:
I see that for serialization we assume the value passed in is not null but for public ctor we initalize them.
![image](https://github.com/microsoft/typespec/assets/64171366/8a0d4465-0bbc-40cd-859f-34adc755d7e0)
This is where the change needs to go since we are constructing List there not change tracking list.  If you want to track this as a separate issue please file / link it here.

Reply:
Opened https://github.com/microsoft/typespec/issues/3509.